### ISSUE TITLE
Explanation needed: ModuleClient.CompleteAsync

### ISSUE THREAD

**Issue by davidguidali** [Azure/azure-iot-sdk-csharp#2148]
**Scenario**

- I have module A which gets data from a server and forwards it to module B which forwards it to IoT Hub
- I don't use any Cloud-To-Device messages

On module B I use SetInputMessageHandlerAsync on the module client and pass over a callback which return MessageResponse.Completed after sending the message to the IoT Hub and MessageResponse.None when there's an error during the sending of the message.

Now I came across the _ModuleClient.CompleteAsync_ method, which is described as:

_Deletes a received message from the module queue_

**Question**
What is the "module queue" if I call this method in module B?
 I guess in my scenario the message is queued on the edge hub when sending from module A to module B and from module B to IoT Hub. 

Or does _ModuleClient.CompleteAsync_ only has to be used in Cloud-to-Device messages to remove the message from the IoT Hub queue?

So basically, what happens if I call this method after sending a message from module B? Its possible, my code runs with or without calling this method.. just wondering if there are some cases where its needed or if it somehow will fill up a queue.

**Comment by timtay-microsoft** [comment]
CompleteAsync behaves like returning MessageResponse.Completed from your message handler. 

Since the message handler has this logic baked in, and since modules cannot receive "cloud to module" messages the way that a device can, there is no reason for you to need to call CompleteAsync from your module client instance.

Does that answer your question?

**Comment by davidguidali** [comment]
Yes, thank you very much!
